File: backend/src/main/java/edu/cit/amihan/medibook/patient/dto/PatientResponse.java
package edu.cit.amihan.medibook.patient.dto;

import edu.cit.amihan.medibook.patient.entity.Patient;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PatientResponse {

    private Long patientId;
    private String fullName;
    private LocalDate dateOfBirth;
    private String contactNumber;
    private String address;
    private String email;

    public static PatientResponse fromEntity(Patient patient) {
        return PatientResponse.builder()
                .patientId(patient.getPatientId())
                .fullName(patient.getFullName())
                .dateOfBirth(patient.getDateOfBirth())
                .contactNumber(patient.getContactNumber())
                .address(patient.getAddress())
                .email(patient.getUser().getEmail())
                .build();
    }
}

File: mobile/app/src/main/java/edu/cit/amihan/medibook/feature/patient/model/PatientResponse.kt
package edu.cit.amihan.medibook.feature.patient.model

import com.google.gson.annotations.SerializedName

data class PatientResponse(
    @SerializedName("patientId") val patientId: Long = 0L,
    @SerializedName("fullName") val fullName: String? = null,
    @SerializedName("dateOfBirth") val dateOfBirth: String? = null,
    @SerializedName("contactNumber") val contactNumber: String? = null,
    @SerializedName("address") val address: String? = null,
    @SerializedName("email") val email: String? = null
)
